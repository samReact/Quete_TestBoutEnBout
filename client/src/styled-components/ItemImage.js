import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  width: 100%;
  padding-top: 100%;
`;

export const Centering = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-content: center;
`;

export const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
  margin-bottom: 48px;
`;
