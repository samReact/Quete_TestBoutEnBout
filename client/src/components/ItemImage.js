import React from 'react';
import propTypes from 'prop-types';
import * as StyledImage from '../styled-components/ItemImage';

const ItemImage = ({ imageUrl }) => (
  <StyledImage.Container>
    <StyledImage.Centering>
      <StyledImage.Image src={imageUrl} alt="Item" />
    </StyledImage.Centering>
  </StyledImage.Container>
);

ItemImage.propTypes = {
  imageUrl: propTypes.string.isRequired,
};

export default ItemImage;
