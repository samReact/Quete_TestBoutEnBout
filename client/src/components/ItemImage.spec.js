import React from 'react';
import { shallow } from 'enzyme';
import ItemImage from './ItemImage';

describe('ItemImage', () => {
  describe('when passed prop imageUrl', () => {
    it('should render properly', () => {
      const wrapper = shallow(<ItemImage imageUrl="http://ok.com/whatever" />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should  has an url', async () => {
      await expect(page).toMatchElement('url');
    });
  });
});
