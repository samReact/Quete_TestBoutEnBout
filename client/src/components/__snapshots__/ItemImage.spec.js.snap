// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemImage when passed prop imageUrl should render properly 1`] = `
<styled.div>
  <styled.div>
    <styled.img
      alt="Item"
      src="http://ok.com/whatever"
    />
  </styled.div>
</styled.div>
`;
